python/lsst/pipe/base/configOverrides.py

```diff
@@ -239,7 +239,7 @@ def applyTo(self, config):
         for otype, override in self._overrides:
             if otype is OverrideTypes.File:
                 with override.open("r") as buffer:
-                    config.loadFromStream(buffer, filename=str(override))
+                    config.loadFromStream(buffer, filename=override.ospath)
```

Comment:
Outside the scope of this, but we should consider implementing `__fspath__` to make ResourcePaths adhere to the `PathLike` protocol. Likely we would default that method to `.ospath`, but it would move the exception to `__fspath__` being called. This would give us the option to implement other behavior at some point in the future (like automatically caching or something) and not have to hunt for all the locations we did `.ospath`

Reply:
Ticketed as [DM-34419](https://jira.lsstcorp.org/browse/DM-34419).